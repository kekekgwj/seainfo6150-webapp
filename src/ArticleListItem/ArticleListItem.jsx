import React,  { useState }  from "react";
import { Link } from "react-router-dom";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import style from "./ArticleListItem.module.css"
const ArticleListItem = (props) => {
    const { article } = props;
    const { 
            slug,
            title,
            timeStamp,
            displayDate,
            shortText
        } = article;
    const [buttonText, setButtonText] = useState('Show more');
    const [showDetail, setShowDetail]  = useState(false);
    const handleButtonClick = () => {
        if (!showDetail) {
            setButtonText('Show less');
        }
        else {
            setButtonText('Show more');
        }
        setShowDetail(!showDetail);

    }
    return (    
    <div className={style.ArticleItemWrapper}>
        <article>
            <header>
                <h1>
                    <Link to={`/articlelist/${slug}`} className={style.Title}>
                        {title}
                    </Link>
                </h1>
                {showDetail && 
                <div>
                    <time className={style.Date} dateTime={timeStamp}>{displayDate}</time>
                     <p>{shortText}</p>
                </div>}  
            </header>
            <ArticleTextToggleButton onClick={handleButtonClick} buttonText={buttonText}></ArticleTextToggleButton>
        </article>
    </div>)
}
export default ArticleListItem;
