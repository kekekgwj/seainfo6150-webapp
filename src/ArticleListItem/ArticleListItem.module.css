.ArticleItemWrapper {
    background-color: white;
    border-radius: 10px;
    padding: 0px 0 10px 10px;
    margin-top: 20px;
    background: linear-gradient(to right, RGB(176,23,31,0.5) 0px, white 20%)
}

.Title {
    color: red;
}

.Date {
    font-weight: bold;
}