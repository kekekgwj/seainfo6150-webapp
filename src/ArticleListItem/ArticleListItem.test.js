import React from "react";
import ArticleListItem from "./ArticleListItem";

describe("ArticleListItem tests", () => {
    it("renders correctly", () => {
        const { container } = render(<ArticleListItem article={{}}/>)
        expect(container).toMatchSnapshot();
    })
})