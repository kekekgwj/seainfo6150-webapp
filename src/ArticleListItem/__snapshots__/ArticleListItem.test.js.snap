// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="ArticleItemWrapper"
  >
    <article>
      <header>
        <div
          class="BasicInfo"
        >
          <div
            class="articleImageWrapper"
          >
            <img
              alt=""
              class="articleImage"
            />
          </div>
          <h1
            class="Header"
          >
            <a
              class="Title"
              href="/articlelist/undefined"
            />
          </h1>
        </div>
      </header>
      <div
        class="ToggleButtonContainer"
      >
        <div
          class="ToggleButtonWrapper"
        >
          <button
            class="ToggleButton"
          >
            Show more
          </button>
        </div>
      </div>
    </article>
  </div>
</div>
`;
